src/job.js

```diff
@@ -0,0 +1,110 @@
+import EventObserver from './events';
+import { protobuf, unserialize } from './protobuf';
+import { CYCLE_STATUS_ACCEPTED, CYCLE_STATUS_REJECTED } from './_constants';
+import { GRID_UNKNOWN_CYCLE_STATUS } from './_errors';
+
+export default class Job {
+  constructor({ worker, modelId, modelVersion, gridClient, logger }) {
+    this.worker = worker;
+    this.modelId = modelId;
+    this.modelVersion = modelVersion;
+    this.grid = gridClient;
+    this.plans = {};
+    this.protocols = {};
+    // holds request_key
+    this.cycleParams = {};
+    this.clientConfig = {};
+    this.logger = logger;
+    this.observer = new EventObserver();
+  }
+
+  on(event, handler) {
+    this.observer.subscribe(event, handler);
+  }
+
+  async initCycle(cycleParams) {
+    this.logger.log(
+      `Cycle initialization with params: ${JSON.stringify(cycleParams)}`
+    );
+    this.cycleParams = cycleParams;
+    this.clientConfig = cycleParams.client_config;
+
+    // load all plans
+    for (let planName of Object.keys(cycleParams.plans)) {
+      const planId = cycleParams.plans[planName];
+      const planBinary = await this.grid.getPlan(
+        this.worker.id,
+        cycleParams.request_key,
+        planId
+      );
+      this.plans[planName] = unserialize(
+        this.worker,
+        planBinary,
+        protobuf.syft_proto.messaging.v1.Plan
+      );
+    }
+
+    // load all protocols
+    for (let protocolName of Object.keys(cycleParams.protocols)) {
+      const protocolId = cycleParams.protocols[protocolName];
+      const protocolBinary = await this.grid.getProtocol(
+        this.worker.id,
+        cycleParams.request_key,
+        protocolId
+      );
+      this.protocols[protocolName] = unserialize(
+        this.worker,
+        protocolBinary,
+        protobuf.syft_proto.messaging.v1.Protocol
+      );
+    }
+  }
+
+  async start() {
+    // request cycle

```

Comment:
`start()`
 - Authentication and receive a worker_id
 - Connection speed test
 - FL Cycle Request
 - ** If accepted, you’ll receive an ID for the model, plans, and protocols AND the full client_config
 - ** If rejected, you’ll receive a timestamp for when start() should be called again
 - If accepted, begin downloading all resources

`on('accepted')`
 - Fires when all resources from a FL cycle request have been completed… meaning that you’ve downloaded the model, all plans, and all protocols successfully